### PR TITLE
Create WebsocketWebClient

### PR DESCRIPTION
**Pull request by KoalaSat**
## What does this PR do?
Part of https://github.com/Reckless-Satoshi/robosats/issues/261

This PR only moves the WebSockets code to a client so it can be tested isolated on the Web fronted.

## Checklist before merging
- [x] If it's a frontend feature, I have ran prettier `cd frontend; npm run format`. If it's a mobile app feature I ran `cd mobile; npm run format`.

### CODE REVIEW

File: frontend/src/components/EncryptedChat.js [hunk 1]
```diff
@@ -52,155 +54,143 @@ class Chat extends Component {
...
-                encryptedMessage: dataFromServer.message,
-                plainTextMessage: dataFromServer.message,
-                validSignature: false,
+                index: dataFromServer.index,
+                encryptedMessage: dataFromServer.message.split('\\').join('\n'),
+                plainTextMessage: decryptedData.decryptedMessage,
+                validSignature: decryptedData.validSignature,
                 userNick: dataFromServer.user_nick,
-                time: new Date().toString(),
+                time: dataFromServer.time,
```
Comment (KoalaSat) [inline frontend/src/components/EncryptedChat.js:163]: I just moved code to a function, github got crazy with the changes to display

File: frontend/src/components/EncryptedChat.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ import { saveAsJson } from '../utils/saveFile';
 import { AuditPGPDialog } from './Dialogs';
 import RobotAvatar from './Robots/RobotAvatar';
 import { systemClient } from '../services/System';
+import { websocketClient } from '../services/Websocket';
```
Comment (Reckless-Satoshi) [inline frontend/src/components/EncryptedChat.js:23]: Legacy unused import `import ReconnectingWebSocket from 'reconnecting-websocket';`

File: frontend/src/services/Websocket/WebsocketConnectionWeb/index.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import ReconnectingWebSocket from 'reconnecting-websocket';
+import { StringOrBuffer } from 'simple-plist';
```
Comment (Reckless-Satoshi) [inline frontend/src/services/Websocket/WebsocketConnectionWeb/index.ts:2]: Unused import

File: frontend/src/components/EncryptedChat.js (hunk 1 above)
Comment (Reckless-Satoshi) [reply frontend/src/components/EncryptedChat.js:163]: I'm certainly having a hard time getting my head around what has changed :smile: . On my tests the Chat does not work. Although the console logs "connected!", the `state.connected` remains `false` . I am sorry I am not able to debug this further at the moment.

File: frontend/src/components/EncryptedChat.js [hunk 4]
```diff
@@ -43,6 +44,7 @@ class Chat extends Component {
     messages: [],
     value: '',
     connected: false,
+    connection: null,
```
Comment (Reckless-Satoshi) [inline frontend/src/components/EncryptedChat.js:47]: New but unused state variable.

File: frontend/src/components/EncryptedChat.js (hunk 1 above)
Comment (Reckless-Satoshi) [reply frontend/src/components/EncryptedChat.js:163]: I'm certainly having a hard time getting my head around what has changed smile . On my tests the Chat does not work. Although the console logs "connected!", the state.connected remains false . I am sorry I am not able to debug this further at the moment.

File: frontend/src/components/EncryptedChat.js (hunk 2 above)
Comment (KoalaSat) [reply frontend/src/components/EncryptedChat.js:23]: no no, this is the new `websocketClient` : https://github.com/Reckless-Satoshi/robosats/pull/277/files#diff-6d4caf7a275f6bc3f26d740379aa63a0d2cddbcbccf44e6611f909af0e68793cR14

File: frontend/src/components/EncryptedChat.js (hunk 4 above)
Comment (KoalaSat) [reply frontend/src/components/EncryptedChat.js:47]: oh that's the problem, I don't how at some point it changes from `this.state.connection.send` to  `this.state.send`, I'll move it back, the idea of this object is to be the one who keep the connection with the server socket

Comment (KoalaSat) [reply frontend/src/components/EncryptedChat.js:47]: https://github.com/Reckless-Satoshi/robosats/pull/277/files#diff-cf6fbea242f7b24c53dea696717984a0fd76bde01795100ae85a8d44c26ac05eR78

**Comment by KoalaSat** [comment]
Tested and working now @Reckless-Satoshi

File: frontend/src/components/EncryptedChat.js (hunk 2 above)
Comment (Reckless-Satoshi) [reply frontend/src/components/EncryptedChat.js:23]: Haha, sorry for the confusion, I could not leave a comment on a line that had not changed. I was pointing out that there is an unused import a few lines before the one I commented.

**Comment by Reckless-Satoshi** [comment]
Briefly tested. Seems to work as expected. Merging!

**Pull request merged**